skt/state_file.py

```diff
@@ -16,6 +16,39 @@
 import os
 
 
+def get_state(state_file, state_key):
+    """
+    Read and return a value from the state file for a specified key.
+
+    Args:
+        state_file: Path to a state file.
+        state_key:  The key for a desired value in the state file.
+
+    Returns:
+        Value from the state file for the corresponding key. The value type
+        will be the same as what was set when the value was stored.
+
+    """
+    config = ConfigParser.RawConfigParser()
+
+    # Does this state file exist?
+    if not os.path.isfile(state_file):
+        return None
+
+    # Read the state file
+    config.read(state_file)
+
+    # Check if the state file has a 'state' section.
+    if not config.has_section("state"):
+        return None
+
+    # Get the value from the state file.
+    try:
+        return config.get('state', state_key)
```

Comment:
Are you sure this piece of code doesn't catch the previous if?

Reply:
So if the `[state]` section exists, but the key we are looking for does not, a `NoOptionError` exception is raised. I caught the exception in that `try:` block to make it easier to handle elsewhere in the package.